### PR TITLE
Implement tests for NooBaa component rescheduling improvements

### PR DESCRIPTION
**Pull request by sidhant-agrawal**
- Add new tests in mcg/test_host_node_failure.py
 - Reduce timeout from 300 seconds to 90 seconds in existing mcg/test_mcg_resources_disruptions.py

Signed-off-by: Sidhant Agrawal <sagrawal@redhat.com>

### CODE REVIEW

File: tests/manage/mcg/test_host_node_failure.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
+import logging
```
Comment (ebenahar) [inline tests/manage/mcg/test_host_node_failure.py:1]: thinking maybe to either put this test file under tests/manage/mcg or keep the current location but just rename with a name that reflects it's a noobaa related test file

Comment (sidhant-agrawal) [reply tests/manage/mcg/test_host_node_failure.py:1]: @ebenahar yes, this test file is already under `tests/manage/mcg`. Do we still need to rename it?

File: tests/manage/mcg/test_host_node_failure.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+from ocs_ci.utility.utils import TimeoutSampler
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@tier4a
+@bugzilla("1853638")
+@ignore_leftovers
+@skipif_vsphere_ipi
```
Comment (ebenahar) [inline tests/manage/mcg/test_host_node_failure.py:36]: why do we skip on vsphere IPI?

File: tests/manage/mcg/test_host_node_failure.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            )
+
+        # Wait for rescheduled pod to reach Running state.
+        # For noobaa-db pod which is attached to a PV it may take more time (~8 minutes)
+        # until the new pod can attach to the PV
+        pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=self.labels_map[noobaa_sts],
+            resource_count=1,
+            timeout=600 if noobaa_sts == "noobaa_db" else 60,
```
Comment (ebenahar) [inline tests/manage/mcg/test_host_node_failure.py:182]: maybe replace "noobaa_db" with a constant

File: tests/manage/mcg/test_host_node_failure.py (hunk 2 above)
Comment (sidhant-agrawal) [reply tests/manage/mcg/test_host_node_failure.py:36]: Test involving node related operations currently fails on vsphere IPI. 
This marker was added in many tests for fixing the issue #4012

**Review by ebenahar — APPROVED** [review]

**Review by MeridianExplorer — APPROVED** [review]

**Pull request merged**